src/components/profile/Profile.js

```diff
@@ -1,51 +1,53 @@
-import React, { Component } from 'react'
-import { connect } from 'react-redux'
-
-import { withRouter } from 'react-router-dom'
+import React from 'react'
+import { useSelector } from 'react-redux'
 
 import PageContainer from '../common/PageContainer';
 import ProfileDetails from './ProfileDetails'
 import ProfileSection from './ProfileSection'
-import ProfileYourKeys from './ProfileYourKeys'
-import ProfileNotice from './ProfileNotice'
 import ProfileQRCode from './ProfileQRCode';
-
-class Profile extends Component {
-   state = {
-      loader: false
-   }
-
-   componentDidMount() {}
-
-   render() {
-      const { account } = this.props
-
-      return (
-         <PageContainer
-            title={`Account: @${account.accountId ? account.accountId : ``}`}
-         >
+import { Wallet } from '../../utils/wallet'
+
+const wallet = new Wallet()
+
+export function Profile({ match }) {
+    const { accountId } = match.params
+    const currentUser = useSelector(state => state.account)
+    const [ account, setAccount ] = React.useState(
+        currentUser.accountId === accountId && currentUser
+    )
+    const [ loading, setLoading ] = React.useState(!account)
+
+    React.useEffect(() => {
+        if (currentUser.accountId === accountId) {
+            setAccount(currentUser)
+        } else {
+            setLoading(true)
+            wallet.getAccount(accountId).state()
```

Comment:
I'd prefer we load through Redux as discussed before. In this specific example these reasons come in play (in addition to consistency / debugging possibility):
1) it'll allow us to cache results globally, so that on next call we can already show data and update it vs load from scratch
2) we won't need boilerplate for triggering loading flag on/off

Reply:
This means that we will add items to localStorage that potentially get very stale, with no plan to clear them. For the foreseeable future this will probably cause no problems, but it seems like transient data and I like that the current implementation treats it as such.

Reply:
I mean just cache in memory, not in local storage.